### PR TITLE
Upgrade OpenLayers

### LINKED ISSUE #15296: Upgrade OpenLayers
Currently phpMyAdmin is using [OpenLayers](https://openlayers.org/) version 2.13.1, that was released in 2013.

We should upgrade OpenLayers to the latest version (which at the time of writing is version 5.3.0).

Related to #13173

### PR DESCRIPTION
**Pull request by shucon**
Signed-off-by: Saksham Gupta <shucon01@gmail.com>

Fixes: https://github.com/phpmyadmin/phpmyadmin/issues/15296

### Description

Please describe your pull request.

Fixes #

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: js/table/gis_visualization.js [hunk 1]
```diff
@@ -148,16 +148,17 @@ function initGISVisualization () {
...
-        OpenLayers._getScriptLocation = function () {
-            return './js/vendor/openlayers/';
-        };
-        // Adds necessary styles to the div that coontains the openStreetMap
+    // Loads a set of small JS file needed for the OpenLayers
+    var head = document.getElementsByTagName('head')[0];
+    var script= document.createElement('script');
+    script.type = 'text/javascript';
+
+    script.src = 'js/vendor/openlayers/OpenLayers.js';
```
Comment (mauriciofauth) [inline js/table/gis_visualization.js:156]: Is this necessary? This script tag is created in:
https://github.com/phpmyadmin/phpmyadmin/blob/bcf21c4081232df1bb77a076a3ff551ee7d7b40c/libraries/classes/Controllers/Table/GisVisualizationController.php#L111

File: libraries/classes/Gis/GisGeometry.php [hunk 2]
```diff
@@ -139,15 +139,20 @@ abstract class GisGeometry
...
+//        . intval($srid) . '"), map.getProjectionObject())); '
+//        . 'bound.extend(new OpenLayers.LonLat('
+//        . $scale_data['maxX'] . ', ' . $scale_data['maxY']
+//        . ').transform(new OpenLayers.Projection("EPSG:'
+//        . intval($srid) . '"), map.getProjectionObject()));';
+        return 'var minLoc = [' . $scale_data['minX'].', '. $scale_data['minY']. ' ];'
+        . 'var maxLoc = [' . $scale_data['maxX'].', '. $scale_data['maxY']. ' ];'
+        . 'var ext = ol.extent.boundingExtent([minLoc, maxLoc]);'
+        . 'ext  = ol.proj.transformExtent(ext, ol.proj.get("EPSG:' . intval($srid) . '"), ol.proj.get(\'EPSG:3857\'));'
+        . 'map.getView().fit(ext, map.getSize());';
```
Comment (mauriciofauth) [inline libraries/classes/Gis/GisGeometry.php:155]: Use `sprintf` here to improve readability.

Comment (shucon) [reply libraries/classes/Gis/GisGeometry.php:155]: I['m not sure why `sprintf` is required here. It's returning a piece of javascript code.

Comment (mauriciofauth) [reply libraries/classes/Gis/GisGeometry.php:155]:
```suggestion
        return sprintf(
            'var minLoc = [%s, %s];'
            . 'var maxLoc = [%s, %s];'
            . 'var ext = ol.extent.boundingExtent([minLoc, maxLoc]);'
            . 'ext = ol.proj.transformExtent(ext, ol.proj.get("EPSG:%s"), ol.proj.get(\'EPSG:3857\'));'
            . 'map.getView().fit(ext, map.getSize());',
            $scale_data['minX'],
            $scale_data['minY'],
            $scale_data['maxX'],
            $scale_data['maxY'],
            intval($srid)
        );
```

Comment (shucon) [reply libraries/classes/Gis/GisGeometry.php:155]: Ah, my bad! I thought `sprintf` was the one used for translations.

File: libraries/classes/Gis/GisGeometry.php [hunk 3]
```diff
@@ -418,9 +403,9 @@ protected function getPointsArrayForOpenLayers(array $points_arr, $srid)
      */
     protected function getPointForOpenLayers(array $point, $srid)
     {
-        return '(new OpenLayers.Geometry.Point(' . $point[0] . ',' . $point[1] . '))'
-        . '.transform(new OpenLayers.Projection("EPSG:'
-        . intval($srid) . '"), map.getProjectionObject())';
+        return '(new ol.geom.Point([' . $point[0] . ',' . $point[1] . '])'
+        . '.transform(ol.proj.get("EPSG:' . intval($srid) . '")'
```
Comment (williamdes) [inline libraries/classes/Gis/GisGeometry.php:407]:
```suggestion
        . '.transform(ol.proj.get("EPSG:' . ((int) $srid) . '")'
```

File: libraries/classes/Gis/GisGeometry.php [hunk 4]
```diff
@@ -418,9 +403,9 @@ protected function getPointsArrayForOpenLayers(array $points_arr, $srid)
...
+        . ', ol.proj.get(\'EPSG:3857\')))';
```
Comment (williamdes) [inline libraries/classes/Gis/GisGeometry.php:408]: is it normal we force 3857 here ?

Comment (shucon) [reply libraries/classes/Gis/GisGeometry.php:408]: Yes, that's the projection system used by OpenLayers is 3857. Check: https://epsg.io/3857

**Comment by shucon** [comment]
Closing in reference to: https://github.com/phpmyadmin/phpmyadmin/pull/16303

**Pull request closed without merging**